cmake/find_modules/FindMySQL.cmake

```diff
@@ -28,8 +28,17 @@ if (DEFINED VCPKG_TARGET_TRIPLET)
   endif()
 
   if (${FOUND_VAR})
-    set(MySQL_FOUND TRUE)
     add_library(MySQL::MySQL ALIAS ${LIBRARY_TARGET})
+
+    get_target_property(INCLUDE_DIRS ${LIBRARY_TARGET} INTERFACE_INCLUDE_DIRECTORIES)
+    if (NOT INCLUDE_DIRS)
+      message(FATAL_ERROR "Expected include paths to be set")
+    endif()
+    foreach(current IN LISTS INCLUDE_DIRS)
+      # vcpkg expects one to include mysql/mysql.h, which goes against how MySQL sets
+      # itself up, so we have to correct for that
+      target_include_directories(${LIBRARY_TARGET} SYSTEM INTERFACE "${current}/mysql")
```

Comment:
Sorry, this doesn't make sense to me: if you need to include `mysql/mysql.h`, then the include directory name should _not_ include the `mysql` path segment, but here we're adding one. Am I missing something or is either the comment or the code wrong?

Reply:
Neither. The comment is correct in stating that vcpkg sets up the include path so that you need to include `mysql/mysql.h` but we don't do it that way (and neither do other means of obtaining MySQL). Hence, we need to add `mysql` to the include path so that we can include `mysql.h` directly.

Reply:
Thanks for the explanation, this was a bit confusing but I see it now.

Reply:
Let me try to rephrase the comment to make it more clear...